runtime/parachains/src/paras_inherent.rs

```diff
@@ -252,25 +256,24 @@ pub mod pallet {
 			// (`enter`) and the off-chain checks by the block author (this function). Once we are confident
 			// in all the logic in this module this check should be removed to optimize performance.
 
-			let inherent_data =
-				match Self::enter(frame_system::RawOrigin::None.into(), inherent_data.clone()) {
-					Ok(_) => inherent_data,
-					Err(err) => {
-						log::error!(
-							target: LOG_TARGET,
-							"dropping paras inherent data because they produced \
+			let inherent_data = match Self::enter_inner::<true>(inherent_data.clone()) {
+				Ok(_) => inherent_data,
+				Err(err) => {
+					log::error!(
+						target: LOG_TARGET,
+						"dropping paras inherent data because they produced \
 							an invalid paras inherent: {:?}",
-							err.error,
-						);
+						err.error,
+					);
 
-						ParachainsInherentData {
-							bitfields: Vec::new(),
-							backed_candidates: Vec::new(),
-							disputes: Vec::new(),
-							parent_header: inherent_data.parent_header,
-						}
-					},
-				};
+					ParachainsInherentData {
+						bitfields: Vec::new(),
+						backed_candidates: Vec::new(),
+						disputes: Vec::new(),
+						parent_header: inherent_data.parent_header,
+					}
+				},
+			};
```

Comment:
At what point will we remove the "training wheels" and no longer call `enter_inner`?

Reply:
Once we refactored this entire thing, I am not entirely happy with it - it's too complex and things are very cross dependent.